File: src/IKVM.Tests/Java/java/lang/SystemTests.cs
using System;
using System.Runtime.InteropServices;
using System.Threading;

using FluentAssertions;

using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace IKVM.Tests.Java.java.lang
{

    [TestClass]
    public class SystemTests
    {

        [TestMethod]
        public void OsNameProperty()
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                global::java.lang.System.getProperty("os.name").Should().StartWith("Windows");
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
                global::java.lang.System.getProperty("os.name").Should().StartWith("Linux");
            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                global::java.lang.System.getProperty("os.name").Should().StartWith("Mac OS X");
        }

        [TestMethod]
        public void CanWriteToStdOut()
        {
            global::java.lang.System.@out.println("TEST");
        }

        [TestMethod]
        public void CanWriteToStdErr()
        {
            global::java.lang.System.err.println("TEST");
        }

        [TestMethod]
        public void NanoTimeShouldReportAtLeastASecond()
        {
            var startTime = global::java.lang.System.nanoTime();
            Thread.Sleep(TimeSpan.FromSeconds(1.1));
            var totalTime = global::java.lang.System.nanoTime() - startTime;
            totalTime.Should().BeGreaterOrEqualTo(1000000000L);
        }

    }

}

File: src/IKVM.Java.Tests/java/lang/SystemTests.java
package ikvm.tests.java.java.lang;

import java.lang.*;
import java.util.*;

@cli.Microsoft.VisualStudio.TestTools.UnitTesting.TestClassAttribute.Annotation()
public class SystemTests {
    
    @cli.Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute.Annotation()
    public void canPrintLnString() throws Throwable {
        System.out.println("tests");
    }
    
    @cli.Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute.Annotation()
    public void canPrintLnObject() throws Throwable {
        System.out.println(new Object());
    }

}
